torch/utils/data/_utils/worker.py

```diff
@@ -132,9 +132,10 @@ def _worker_loop(dataset_kind, dataset, index_queue, data_queue, done_event,
 
         try:
             if init_fn is not None:
-                init_fn(worker_id)
+                init_fn(worker_id, dataset)
```

Comment:
This can absolutely not happen. You need BC-preserving mechanism here. Also, why do you even need this? You can get the dataset object from `torch.utils.data.get_worker_info`

Reply:
In the context of ChunkDataset on multiple workers, each worker needs to have access to the dataset, so it can configure its internal data reading engine. Without that, each worker would read the same data.

`torch.utils.data.get_worker_info` is a good idea! The dataset argument was presenting the original PR, before the new DataLoader implementation - I forgot to use it. Thanks for pointing it out!

Reply:
Agreed. Using `torch.utils.data.get_worker_info().dataset`